subsys/bluetooth/controller/ll_sw/nordic/lll/lll_clock.c

```diff
@@ -19,61 +19,75 @@
 #define LFCLOCK_TIMEOUT_MS 500
 #define HFCLOCK_TIMEOUT_MS 2
 
-static void clock_ready(struct device *dev, clock_control_subsys_t subsys,
-			void *user_data);
+struct lll_clock_state {
+	struct onoff_client cli;
+	struct k_sem sem;
+};
 
-static struct device *dev;
+static struct lll_clock_state lf_state;
+static struct lll_clock_state hf_state;
+static atomic_val_t hf_refcnt;
 
-int lll_clock_init(void)
+static void clock_ready(struct onoff_manager *mgr, struct onoff_client *cli,
+			uint32_t state, int res)
+{
+	struct lll_clock_state *clk_state =
+		CONTAINER_OF(cli, struct lll_clock_state, cli);
+
+	k_sem_give(&clk_state->sem);
+}
+
+int clk_request(struct onoff_manager *mgr, struct lll_clock_state *state)
```

Comment:
This should be `static` as it's currently a global symbol.

Reply:
done